### PR TITLE
Feature/add profile tests

### PR DESCRIPTION
**Pull request by NathanDVT**
- [X] Added Profile Unit test
- [X] Added Profile UI test
- [X] Added Fastlane skeleton

### CODE REVIEW

File: realwork/fastlane/SnapshotHelper.swift [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+            guard self.app != nil else {
+                NSLog("XCUIApplication is not set. Please call setupSnapshot(app) before snapshot().")
+                return
+            }
+
+            let screenshot = XCUIScreen.main.screenshot()
+            guard var simulator = ProcessInfo().environment["SIMULATOR_DEVICE_NAME"], let screenshotsDir = screenshotsDirectory else { return }
+
+            do {
+                // The simulator name contains "Clone X of " inside the screenshot file when running parallelized UI Tests on concurrent devices
```
Comment (NathanDVT) [inline realwork/fastlane/SnapshotHelper.swift:180]: Codacy Issue found: [Line should be 120 characters or less: currently 144 characters](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5389722)

**Comment by NathanDVT** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 8
           

Clones removed
==============
+ realwork/realworkTests/ProfileTest/ProfileViewModelTest.swift  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5389722&bid=17669953)

File: realwork/fastlane/SnapshotHelper.swift [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+            return
+        }
+
+        let path = cacheDirectory.appendingPathComponent("snapshot-launch_arguments.txt")
+        app.launchArguments += ["-FASTLANE_SNAPSHOT", "YES", "-ui_testing"]
+
+        do {
+            let launchArguments = try String(contentsOf: path, encoding: String.Encoding.utf8)
+            let regex = try NSRegularExpression(pattern: "(\\\".+?\\\"|\\S+)", options: [])
+            let matches = regex.matches(in: launchArguments, options: [], range: NSRange(location: 0, length: launchArguments.count))
```
Comment (NathanDVT) [inline realwork/fastlane/SnapshotHelper.swift:141]: Codacy Issue found: [Line should be 120 characters or less: currently 133 characters](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5389722)

File: realwork/fastlane/SnapshotHelper.swift [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+            app.typeKey(XCUIKeyboardKeySecondaryFn, modifierFlags: [])
+        #else
+
+            guard self.app != nil else {
+                NSLog("XCUIApplication is not set. Please call setupSnapshot(app) before snapshot().")
+                return
+            }
+
+            let screenshot = XCUIScreen.main.screenshot()
+            guard var simulator = ProcessInfo().environment["SIMULATOR_DEVICE_NAME"], let screenshotsDir = screenshotsDirectory else { return }
```
Comment (NathanDVT) [inline realwork/fastlane/SnapshotHelper.swift:177]: Codacy Issue found: [Line should be 120 characters or less: currently 143 characters](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5389722)

File: realwork/fastlane/SnapshotHelper.swift [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+            NSLog("Couldn't detect/set launch_arguments...")
+        }
+    }
+
+    open class func snapshot(_ name: String, timeWaitingForIdle timeout: TimeInterval = 20) {
+        if timeout > 0 {
+            waitForLoadingIndicatorToDisappear(within: timeout)
+        }
+
+        NSLog("snapshot: \(name)") // more information about this, check out https://docs.fastlane.tools/actions/snapshot/#how-does-it-work
```
Comment (NathanDVT) [inline realwork/fastlane/SnapshotHelper.swift:156]: Codacy Issue found: [Line should be 120 characters or less: currently 139 characters](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5389722)

File: realwork/fastlane/SnapshotHelper.swift [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+            return
+        #endif
+
+        guard let app = self.app else {
+            NSLog("XCUIApplication is not set. Please call setupSnapshot(app) before snapshot().")
+            return
+        }
+
+        let networkLoadingIndicator = app.otherElements.deviceStatusBars.networkLoadingIndicators.element
+        let networkLoadingIndicatorDisappeared = XCTNSPredicateExpectation(predicate: NSPredicate(format: "exists == false"), object: networkLoadingIndicator)
```
Comment (NathanDVT) [inline realwork/fastlane/SnapshotHelper.swift:205]: Codacy Issue found: [Line should be 120 characters or less: currently 158 characters](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5389722)

File: realwork/fastlane/SnapshotHelper.swift [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+        return homeDir.appendingPathComponent("Library/Caches/tools.fastlane")
+    }
+}
+
+private extension XCUIElementAttributes {
+    var isNetworkLoadingIndicator: Bool {
+        if hasWhiteListedIdentifier { return false }
+
+        let hasOldLoadingIndicatorSize = frame.size == CGSize(width: 10, height: 20)
+        let hasNewLoadingIndicatorSize = frame.size.width.isBetween(46, and: 47) && frame.size.height.isBetween(2, and: 3)
```
Comment (NathanDVT) [inline realwork/fastlane/SnapshotHelper.swift:245]: Codacy Issue found: [Line should be 120 characters or less: currently 122 characters](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5389722)

File: realwork/fastlane/SnapshotHelper.swift [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+    if waitForLoadingIndicator {
+        Snapshot.snapshot(name)
+    } else {
+        Snapshot.snapshot(name, timeWaitingForIdle: 0)
+    }
+}
+
+/// - Parameters:
+///   - name: The name of the snapshot
+///   - timeout: Amount of seconds to wait until the network loading indicator disappears. Pass `0` if you don't want to wait.
```
Comment (NathanDVT) [inline realwork/fastlane/SnapshotHelper.swift:35]: Codacy Issue found: [Line should be 120 characters or less: currently 126 characters](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5389722)

File: realwork/fastlane/SnapshotHelper.swift [hunk 8]
```diff
@@ -0,0 +1,303 @@
...
+    var debugDescription: String {
+        switch self {
+        case .cannotDetectUser:
+            return "Couldn't find Snapshot configuration files - can't detect current user "
+        case .cannotFindHomeDirectory:
+            return "Couldn't find Snapshot configuration files - can't detect `Users` dir"
+        case .cannotFindSimulatorHomeDirectory:
+            return "Couldn't find simulator home location. Please, check SIMULATOR_HOST_HOME env variable."
+        case .cannotAccessSimulatorHomeDirectory(let simulatorHostHome):
+            return "Can't prepare environment. Simulator home location is inaccessible. Does \(simulatorHostHome) exist?"
```
Comment (NathanDVT) [inline realwork/fastlane/SnapshotHelper.swift:56]: Codacy Issue found: [Line should be 120 characters or less: currently 121 characters](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5389722)

**Comment by NathanDVT** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ realwork/realworkTests/ProfileTest/ProfileViewModelTest.swift  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5389722&bid=17669953)

**Comment by NathanDVT** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ realwork/realworkTests/ProfileTest/ProfileViewModelTest.swift  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5389722&bid=17669953)

**Review by JancoDVTDev — APPROVED** [review]

**Review by JancoDVTDev — APPROVED** [review]
Pull Request is very big. Maybe do UI and Unit tests in seperate features in the future

**Comment by NathanDVT** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ realwork/realworkTests/ProfileTest/ProfileViewModelTest.swift  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5389722&bid=17669953)

**Pull request merged**